### ISSUE TITLE
usb.core.USBError: [Errno 32] Pipe error

### ISSUE THREAD

**Issue by martinayotte** [charlysan/pyrfidhid#5]
Hi Charlysan,
I've install you library and trying to run it failed : 

```
root@orangepipcplus:~/pyrfidhid-master# python examples/read.py 
Initializing device...
Traceback (most recent call last):
  File "examples/read.py", line 76, in <module>
    main()
  File "examples/read.py", line 48, in main
    rfid.init()
  File "build/bdist.linux-armv7l/egg/rfidhid/core.py", line 58, in init
  File "build/bdist.linux-armv7l/egg/rfidhid/usb_hid.py", line 43, in get_report_descriptor
  File "build/bdist.linux-armv7l/egg/usb/control.py", line 173, in get_descriptor
  File "build/bdist.linux-armv7l/egg/usb/core.py", line 1043, in ctrl_transfer
  File "build/bdist.linux-armv7l/egg/usb/backend/libusb1.py", line 883, in ctrl_transfer
  File "build/bdist.linux-armv7l/egg/usb/backend/libusb1.py", line 595, in _check
usb.core.USBError: [Errno 32] Pipe error

```
I've also tried on my own desktop, and it is failing with the same stacktrace ...
Any idea ?

**Comment by charlysan** [comment]
Hi @martinayotte,

A similar issue has been reported by another user [here](https://github.com/charlysan/pyrfidhid/issues/3#issuecomment-414809450). 

- Are you sure your device is the the one described at the [wiki section](https://github.com/charlysan/pyrfidhid/wiki/Reverse-Engineering-A-USB-HID-RFID-Reader-Writer)? 
- Have you checked that your device has this vid:pid pair? `(ffff:0035)`
- Does your device work with IDRW Windows app?

**Comment by lhlong** [comment]
> Hi @martinayotte,
> 
> A similar issue has been reported by another user [here](https://github.com/charlysan/pyrfidhid/issues/3#issuecomment-414809450).
> 
> * Are you sure your device is the the one described at the [wiki section](https://github.com/charlysan/pyrfidhid/wiki/Reverse-Engineering-A-USB-HID-RFID-Reader-Writer)?
> * Have you checked that your device has this vid:pid pair? `(ffff:0035)`
> * Does your device work with IDRW Windows app?

Have same problem.
Hi @charlysan 
I checked on ubuntu/mac and it didn't work.

**Comment by charlysan** [comment]
Hi @lhlong,

I'm almost sure that your device is not supported by my library. I will add a better error handling to show an error message for these cases.
Please provide the specs for your device so I can double check.

**Comment by lhlong** [comment]
Hi @charlysan, 
Thank for your reply.

My device infos:
Bus 001 Device 003: ID 08ff:0009 AuthenTec, Inc.

What is the version of `pyusb` we use?

**Comment by martinayotte** [comment]
In my case : 
Bus 003 Device 002: ID 072f:2200 Advanced Card Systems, Ltd ACR122U

It is working well when using python-pyscard along with pcscd daemon.

**Comment by charlysan** [comment]
@martinayotte,
`ACR122U` is a 13.56MHZ reader, it is not compatible with `pyrfidhid` (125KHZ).

**Comment by jacobcui** [comment]
hi @charlysan I failed on this error too. I've set vid:pid correctly. Here is my lsusb output:

Bus 001 Device 011: ID 6688:6850 CTX 203-ID-RW

**Comment by charlysan** [comment]
@jacobcui, does your device work with IDRW v3 windows app? If not, then you will not be able to use this library, as for the moment it only supports devices that works with that app. I will create a fix though to show a more descriptive error message for those cases.

**Comment by jacobcui** [comment]
Yes my device works with the mentioned Windows app.

**Comment by charlysan** [comment]
@jacobcui, could you please provide the output you get from this command?
```
sudo lsusb -vd 6688:6850
```

Which OS are you using?

A picture of the back of the device might be useful too (like the one I included in the wiki)

**Comment by charlysan** [comment]
@lhlong, sorry for the delay; I missed your last post. I've been using `pyusb 1.0.2`. Same question for you, does your device work with IDRW v3 windows app?

**Comment by charlysan** [comment]
I'm closing this issue as all the errors described above seem to be related to unsupported devices. 
I've included better error handling in version v1.1.4, so you should see now a more verbose error message for any `core.usb.USBError` exceptions.
